src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationService.kt

```diff
@@ -341,30 +341,31 @@ internal class RecommendationService(
     featureFlags: FeatureFlags?,
   ): RecommendationResponse {
     validateRecallType(jsonRequest)
-    val requestJson = JSONObject(jsonRequest.toString())
+    val requestJson = jacksonObjectMapper().readTree(jsonRequest.toString())
     val existingRecommendationEntity = getRecommendationEntityById(recommendationId)
 
-    val sendConsiderationRationaleToDelius = requestJson.optBoolean("sendConsiderationRationaleToDelius")
-    val considerationSensitive = requestJson.optBoolean("considerationSensitive")
+    val sendConsiderationRationaleToDelius = requestJson.has("sendConsiderationRationaleToDelius") &&
+      requestJson.get("sendConsiderationRationaleToDelius").asBoolean()
+    val considerationSensitive = requestJson.has("considerationSensitive") &&
+      requestJson.get("considerationSensitive").asBoolean()
 
-    // remove these from the incoming request as they are not part of the recommendation model, but rather values applies to sending messages to nDelius.
-    val incoming = jsonRequest as ObjectNode
+    // Remove these from the incoming request as they are not part of the recommendation model, but rather values applies to sending messages to nDelius.
+    // Ensure we don't change the 'requestJson' input variable reference otherwise this will affect any possible failure retries
+    val incoming = requestJson.deepCopy() as ObjectNode
```

Comment:
Reading up on `deepCopy` looks like this *should* do the trick and avoid mutation of even deeply nested properties of the parameter object